### PR TITLE
Handling null values for total-count header.

### PR DESCRIPTION
**Pull request by distantcam**
Fix for #664

### CODE REVIEW

**Comment by distantcam** [comment]
@Particular/serviceinsight-maintainers So how do we feel about this one?

**Comment by HEskandari** [comment]
I'd say if we have no idea how this happens and the header is always expected to be there and has been there in every release of SC, there's no reason to write defensive code around it.

If anything, we need to throw a more meaningful exception highlighting the problem, because with no total-count, other stuff like paging are going to get broken as well.

File: src/ServiceInsight/ServiceControl/DefaultServiceControl.cs [hunk 1]
```diff
@@ -236,23 +236,25 @@ IRestClient CreateClient(string baseUrl = null)
...
+            : new RestRequestWithCache(MessagesEndpoint, RestRequestWithCache.CacheStyle.IfNotModified);
 
         PagedResult<T> GetPagedResult<T>(RestRequestWithCache request) where T : class, new()
         {
             var result = Execute<PagedResult<T>, List<T>>(request, response => new PagedResult<T>
             {
                 Result = response.Data,
-                TotalCount = int.Parse(response.Headers.First(x => x.Name == ServiceControlHeaders.TotalCount).Value.ToString())
+                TotalCount = int.Parse(
+                    response.Headers.FirstOrDefault(x => x.Name == ServiceControlHeaders.TotalCount)?.Value?.ToString() ?? "0")
```
Comment (adamralph) [inline src/ServiceInsight/ServiceControl/DefaultServiceControl.cs]: Is this line the only relevant change?

I.e. is the rest of this commit just whitespace fixes?

**Comment by adamralph** [comment]
I tend to agree that we should throw a specific exception when we are not told the total count.

**Comment by distantcam** [comment]
Silently trying to recover from this failure isn't the right solution. Closing this PR.

**Pull request closed without merging**